tools/lib/provision.py

```diff
@@ -247,12 +249,24 @@ def main(options):
         run(["sudo", "service", "redis-server", "restart"])
         run(["sudo", "service", "memcached", "restart"])
     if not options.is_production_travis:
-        # These won't be used anyway
```

Comment:
Good question!

This looks like one of those comments that made sense in the original commit that introduced it, but the context has changed since then and the comment hasn't been adapted to match.

A good technique in general for figuring out this kind of comment is to use `git log` to find the commit that added it, with `-S` (search for `-S` in `git help log`, or in [the web version](https://git-scm.com/docs/git-log) of the same help page):
```
git log --stat -p --follow -S'be used' tools/lib/provision.py
```
(The `--stat -p` is the usual way I read commits, though in this case with one file the `--stat` doesn't add much. The `--follow` lets you trace back transparently through times the whole file was moved around.)

And in fact, in this case, that pulls up commit [5a109cf81](https://github.com/zulip/zulip/commit/5a109cf81). I think in that context the meaning of the comment becomes clear.

This is also a nice illustration of how writing a clear comment can be subtle. :-) A better comment to write in the first place might have been something like "Some steps aren't needed for our test suite, only the dev server. Skip them in Travis."

A pronoun like "these" or "this" is particularly fragile, as it can rapidly become unclear as the code changes what it's referring to.

I think skipping a comment and letting the conditional speak for itself is also good.